lib/definitions/cloud-build-service.d.ts

```diff
@@ -1,42 +1,146 @@
+
+	/**
+	 * Base64 encoded data used for generating QR code image.
+	 */
 	imageData: string;
 }
 
+/**
+ * Defines operations for building a project in the cloud.
+ */
 interface ICloudBuildService {
+	/**
+	 * Builds the specified application in the cloud and returns information about the whole build process.
+	 * @param {IProjectSettings} projectSettings Describes the current project - project dir, application identifier, name and nativescript data.
+	 * @param {string} platform The mobile platform for which the application should be built: Android or iOS.
+	 * @param {string} buildConfiguration The build configuration - Debug or Release.
+	 * @param {IAndroidBuildData} androidBuildData Android speicific information for the build.
+	 * @param {IIOSBuildData} iOSBuildData iOS speicific information for the build.
+	 * @returns {Promise<IBuildResultData>} Information about the build process. It is returned only on successfull build. In case the build fails, the Promise is rejected with the server information.
+	 */
 	build(projectSettings: IProjectSettings,
 		platform: string, buildConfiguration: string,
 		androidBuildData?: IAndroidBuildData,
 		iOSBuildData?: IIOSBuildData): Promise<IBuildResultData>;
+
+	/**
+	 * Validates the build properties for specified platform (Android or iOS).
+	 * The result promise is rejected with the error found. In case everything is correct, the promise is resolved.
+	 * @param {string} platform The mobile platform for which the application should be built: Android or iOS.
+	 * @param {string} buildConfiguration The build configuration - Debug or Release.
+	 * @param {string} projectId Application identifier of the project.
+	 * @param {IAndroidBuildData} androidBuildData Android speicific information for the build.
+	 * @param {IIOSBuildData} iOSBuildData iOS speicific information for the build.
+	 * @returns {Promise<void>}
+	 */
 	validateBuildProperties(platform: string,
 		buildConfiguration: string,
 		projectId: string,
 		androidBuildData?: IAndroidBuildData,
 		iOSBuildData?: IIOSBuildData): Promise<void>;
 }
 
+/**
+ * Describes the project settings required for different operations.
+ */
 interface IProjectSettings {
+	/**
+	 * The directory where the project is located. This should be the path to the directory where application's package.json is located.
+	 */
 	projectDir: string;
+
+	/**
+	 * Application identifier.
+	 */
 	projectId: string;
+
+	/**
+	 * Name that will be visible to the users when application is installed on device.
+	 */
 	projectName: string;
+
+	/**
+	 * The value of `nativescript` key from project's package.json.
+	 */
 	nativescriptData: any;
 }
 
+/**
+ * Describes specific data required for Android Builds.
+ */
 interface IAndroidBuildData {
+	/**
+	 * Path to certificate (.p12 or .keystore), used for code signing the application. Required and used only for release builds.
+	 */
 	pathToCertificate: string;
+
+	/**
+	 * Password of the specified certificate.
```

Comment:
It is obvious, but you could still add that this too is only required and used in release builds